### PR TITLE
Typography updates

### PR DESCRIPTION
**Pull request by soniaturcotte**
- Remove semi-bold and extra-bold font weight
- Add H4 and small body text 
- Update type sizes and remove blue for heading 
- Add link interactive states

Details in this issue: https://github.com/citizensadvice/design-system-testing/issues/1

### CODE REVIEW

File: scss/base/4-elements/_links.scss [hunk 1]
```diff
@@ -1,17 +1,21 @@
 a,
 .hyperlink {
-    color: $cads-palette__black;
+    color: $cads-palette__heritage-blue;
     line-height: 1.25em;
 
     &:visited {
-        color: green;
+        color: #53388D;
```
Comment (panosvoudouris) [inline scss/base/4-elements/_links.scss]: No hex values, use one in the language/palette and if doesn't exist then we have to add it.

File: scss/base/4-elements/_links.scss [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
+        color: #002F55;
+        background-color: $cads-palette__blue-light;
     }
 
     &:active {
-        color: white;
+        color: $cads-palette__black;
+        background-color: $cads-palette__heritage-yellow;
+        text-decoration: none;
+        border-bottom: 4px solid $cads-palette__black;
```
Comment (panosvoudouris) [inline scss/base/4-elements/_links.scss]: Maybe we should add something border width so we don't use px values?

File: scss/base/4-elements/_typography.scss [hunk 3]
```diff
@@ -1,51 +1,82 @@
...
-    font-size: 3em;
-    line-height: 1.25em;
-    font-weight: $cads-font-weight__extra-bold;
-    color: $cads-palette__heritage-blue;
+    font-size: 2.5em;
+    line-height: 1.2em;
+    font-weight: $cads-font-weight__bold;
+    color: $cads-palette__black;
     margin: 0;
     padding: 1em 0 1em 0;
```
Comment (panosvoudouris) [inline scss/base/4-elements/_typography.scss]: Are we leaving the spacing above/below?

Comment (soniaturcotte) [reply scss/base/4-elements/_typography.scss]: Spacing still needs to be worked out / designed so we will add that as a separate pull request.

File: scss/base/4-elements/_links.scss (hunk 2 above)
Comment (soniaturcotte) [reply scss/base/4-elements/_links.scss]: We will leave this for now and see if we want to have it as a variable once we have designed other interactive components.

File: scss/base/4-elements/_links.scss (hunk 1 above)
Comment (soniaturcotte) [reply scss/base/4-elements/_links.scss]: Resolved in most recent commit

**Pull request merged**